napari/layers/labels/labels.py

```diff
@@ -754,34 +754,36 @@ def _lookup_with_index(self, im, selected_label=None):
             Value of selected label to color, by default None
         """
         if selected_label:
-            selected_color = low_discrepancy_image(selected_label, self._seed)
-            if selected_label > len(self._all_vals):
-                self._all_vals = low_discrepancy_image(
-                    np.arange(selected_label + 1), self._seed
+            if selected_label > len(self._all_vals) + 1:
```

Comment:
I think this line is causing your tests to fail - shouldn't it be `selected_label > len(self._all_vals) - 1`?